.github/workflows/src/arm-incremental-typespec-preview.js

```diff
@@ -55,29 +58,61 @@ export default async function incrementalTypeSpec({ github, context, core }) {
     }
   }
 
-  const changedSpecDirs = new Set(
-    changedRmSwaggerFiles.map((f) => dirname(dirname(dirname(f)))),
-  );
+  /** @type Set<string> */
+  const changedReadmeInputFiles = new Set();
+
+  for (const readmeFile of changedRmFiles.filter(readme)) {
+    core.info(`Extracting input files from '${readmeFile}'`);
+
+    let readmeText;
+    try {
+      readmeText = await show("HEAD", readmeFile, core);
```

Comment:
I assume you are doing a git show to avoid checking out the repo. I'm curious how the performance is of this in comparison to just doing the checkout?

Also given this is a form of checking out the untrusted git files we need to be careful about what we doing with these files if we are running with extra permissions. I'm not sure if we are or not but just wanted to call it out.